### PR TITLE
Phase3_Excersis1

### PR DESCRIPTION
**Pull request by Nagalla-QA**

### CODE REVIEW

File: elements/document/collection/nuxeo-collection-create-layout.html [hunk 1]
```diff
@@ -33,6 +33,14 @@
       required
     >
     </nuxeo-input>
+    <nuxeo-input
```
Comment (Gabez0r) [inline elements/document/collection/nuxeo-collection-create-layout.html:36]: Usually we leave a space between widgets.
```suggestion

    <nuxeo-input
```

File: ftest/features/clipboard.feature [hunk 2]
```diff
@@ -55,13 +55,7 @@ Feature: Clipboard
...
      | paste  | 3         |
 
-  Scenario: Can't use clipboard for destinations which don't allow certain types
-    Given I browse to the document with path "/default-domain/Src"
-    And I select the "File1" document
-    And I can see the selection toolbar
-    And I can add selection to clipboard
-    When I browse to the document with path "/default-domain"
-    Then I can see clipboard actions disabled
+
```
Comment (Gabez0r) [inline ftest/features/clipboard.feature]: No need for this extra line.

File: ftest/features/create_doc.feature [hunk 3]
```diff
@@ -1,3 +1,4 @@
+@watch
```
Comment (Gabez0r) [inline ftest/features/create_doc.feature]: No need to push this @watch tag, as this is mostly used for development puproses.

File: elements/document/collection/nuxeo-collection-metadata-layout.html [hunk 4]
```diff
@@ -29,6 +29,10 @@
       <label>[[i18n('label.dublincore.title')]]</label>
       <div name="title">[[document.properties.dc:title]]</div>
     </div>
+    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```
Comment (Gabez0r) [inline elements/document/collection/nuxeo-collection-metadata-layout.html:33]: Please don't forget to leave an empty line of text in between widgets.
```suggestion

    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```

File: elements/document/file/nuxeo-file-metadata-layout.html [hunk 5]
```diff
@@ -29,6 +29,10 @@
       <label>[[i18n('label.dublincore.title')]]</label>
       <div name="title">[[document.properties.dc:title]]</div>
     </div>
+    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```
Comment (Gabez0r) [inline elements/document/file/nuxeo-file-metadata-layout.html:33]: Same here:
```suggestion

    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```

File: elements/document/folder/nuxeo-folder-metadata-layout.html [hunk 6]
```diff
@@ -28,6 +28,10 @@
       <label>[[i18n('label.dublincore.title')]]</label>
       <div name="title">[[document.properties.dc:title]]</div>
     </div>
+    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```
Comment (Gabez0r) [inline elements/document/folder/nuxeo-folder-metadata-layout.html:32]: And here:
```suggestion

    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```

File: elements/document/note/nuxeo-note-metadata-layout.html [hunk 7]
```diff
@@ -28,6 +28,10 @@
       <label>[[i18n('label.dublincore.title')]]</label>
       <div name="title">[[document.properties.dc:title]]</div>
     </div>
+    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```
Comment (Gabez0r) [inline elements/document/note/nuxeo-note-metadata-layout.html:32]: And here:
```suggestion

    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```

File: elements/document/workspace/nuxeo-workspace-metadata-layout.html [hunk 8]
```diff
@@ -28,6 +28,10 @@
       <label>[[i18n('label.dublincore.title')]]</label>
       <div name="title">[[document.properties.dc:title]]</div>
     </div>
+    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```
Comment (Gabez0r) [inline elements/document/workspace/nuxeo-workspace-metadata-layout.html:32]: And here:
```suggestion

    <div role="widget" hidden$="[[!document.properties.dc:source]]">
```

**Review by Gabez0r — COMMENTED** [review]
Good work. Please see comments for:
- development: https://github.com/nuxeo/nuxeo-web-ui/pull/1454#pullrequestreview-1024500921
- functional tests: https://github.com/nuxeo/nuxeo-web-ui/pull/1491#pullrequestreview-1024516164

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 9]
```diff
@@ -247,7 +285,19 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
...
+      }
+    }
+
+    _documentUID(document) {
+      if (document) {
+        return `(${document.uid.substring(document.uid.lastIndexOf('-') + 1, document.uid.length)})`.trim();
+      }
+    }
+
+    _isTrashed(document) {
```
Comment (richardsd) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js:289]: You could do this simpler with 

```suggestion
    _isTrashed(document) {
      return document && document.isTrashed;
    }
```

File: elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+import '@nuxeo/nuxeo-ui-elements/widgets/nuxeo-tooltip.js';
+import '../document/nuxeo-document-form-layout.js';
+import '@polymer/paper-button/paper-button.js';
+
+/**
+`nuxeo-document-parent-inspector-button`
+@group Nuxeo UI
+@element nuxeo-document-parent-inspector-button
+*/
+Polymer({
```
Comment (richardsd) [inline elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js]: Please read Gabriel's comment here about using the old Polymer factory function syntax https://github.com/nuxeo/nuxeo-web-ui/pull/1454#discussion_r910797179

File: elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+        <div class="parentscroll">
+          <div class="scrollable">
+            <div>
+              <h1>Parent Inspector</h1>
+              <div>
+                <p>This is the parent inspector. It displays information about the parent of this document.</p>
+                <table class="table">
+                  <tr>
+                    <td>Tile:</td>
+                    <td>[[document.title]]</td>
```
Comment (richardsd) [inline elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js]: It seems that you are displayed the current document and not the parent. You might need something extra for this to work.

File: elements/nuxeo-web-ui-bundle.html [hunk 12]
```diff
@@ -231,6 +231,16 @@
 
 <!-- DOCUMENT ACTIONS -->
 
+<nuxeo-slot-content name="parentInspectorDocumentAction" slot="DOCUMENT_ACTIONS" order="10">
+  <template>
+    <nuxeo-filter user="[[user]]" expression="user.isAdministrator || this.isMember(user, 'powerusers')">
```
Comment (richardsd) [inline elements/nuxeo-web-ui-bundle.html:236]: As an alternative solution, please check https://github.com/nuxeo/nuxeo-elements/blob/maintenance-3.0.x/ui/nuxeo-filter.js#L167. nuxeo-filter has a group property that can be use to check user membership to groups.

File: ftest/features/clipboard.feature [hunk 13]
```diff
@@ -1,9 +1,8 @@
 @cleanupLocalStorage
 Feature: Clipboard
-
   I can add, remove, move and copy items from/to the clipboard
 
-  Background:
+  Background:
```
Comment (richardsd) [inline ftest/features/clipboard.feature]: nit: extra blank space here.

File: i18n/messages-ar.json [hunk 14]
```diff
@@ -983,6 +983,7 @@
   "label.dublincore.state": "State",
   "label.dublincore.subjects": "Subjects",
   "label.dublincore.title": "Title",
+  "label.dublincore.source": "Source",
```
Comment (richardsd) [inline i18n/messages-ar.json:986]: You only need to add the label to the `messages.json` file. No need to manually add the label to all the translation files, as the translation process happens at Crowdin. Please review the wiki page with details about this mechanism.

**Review by richardsd — COMMENTED** [review]
Continue the good work 🙂

**Pull request closed without merging**